lib/spack/spack/build_systems/cmake.py

```diff
@@ -222,7 +223,7 @@ def define(cmake_var, value):
             value = "ON" if value else "OFF"
         else:
             kind = 'STRING'
-            if isinstance(value, (list, tuple)):
+            if isinstance(value, Sequence) and not isinstance(value, str):
```

Comment:
I think this would be a little more robust:
```suggestion
            if isinstance(value, Sequence) and not isinstance(value, six.string_types):
```
in case we need to `import six` at the top too.

Reply:
Yes, correct. I didn't know six was used (the llnl.util mappings are redundant with capabilities).